src/rule_evaluator.ts

```diff
@@ -78,15 +105,29 @@ function evaluateObfuscatedCondition(
     {},
   );
   const value = hashedSubjectAttributes[condition.attribute];
+  const conditionValueType = targetingRuleConditionValuesTypesFromValues(value);
+
   if (value != null) {
     switch (condition.operator) {
       case getMD5Hash(OperatorType.GTE):
+        if (conditionValueType === OperatorValueType.SEM_VER) {
```

Comment:
not your code, but seems kind of crazy w have to duplicate the switch statement. If easy, consider cleaning it up by changing from a `switch` to a `Map` or something.

Reply:
Aaron do you mean the duplication between the `evaluateCondition` and `evaluateObfuscatedCondition` functions?

Reply:
Yeah (had to expand the PR to confirm). Whenever I see the same business logic change being made in multiple places in a PR like that it triggers some spidey senses 🕷️ 🕸️